### ISSUE TITLE
Saving large playlists might interrupt playback

### ISSUE THREAD

**Issue by kpcee** [DeaDBeeF-Player/deadbeef#2259]
Deadbeef saves the playlist automatically each time it is changed, for larger playlists this can lead to an interruption of the playback.
Examples are moving tracks within the playlist, inserting tracks, deleting tracks from the playlist or disk.
The behavior is therefore justified, since a shutdown of the computer can otherwise result in data loss, see https://github.com/DeaDBeeF-Player/deadbeef/issues/1014 

**Disadvantages:**
* Possible interruptions in playback
* Creates a lot of unnecessary write load, bad for SSDs

A possible solution would be an option in the setting that Deadbeef only saves the changes at the end. This could be done with a single wrapper in the function `int (*plt_save_n) (int n);` which sets plt_modified instead of saving.
The two functions `int (*pl_save_current) (void);` and `int (*plt_save_config) (ddb_playlist_t *plt);` internally use plt_save_n themselves. Existing source code would not need to be further changed.
`int (*pl_save_all) (void);` is not affected and will save the playlists on exit.

**Comment by Alexey-Yakovenko** [comment]
> A possible solution would be an option in the setting that Deadbeef only saves the changes at the end.

Such potential solution contradicts the initial statement that "The behavior is therefore justified, since a shutdown of the computer can otherwise result in data loss".
So this solution is absolutely not acceptable.

**Comment by kpcee** [comment]
This may apply to users who don't quit their programs before the shutdown, but it doesn't affect me and maybe others and it is always a good thing to give the user a choice, especially as the current method is not necessarily good for their own hardware. Furthermore, in my opinion, the user himself is to blame if he does not close the programs correctly. I don't think that there are many programs out there that support such a behavior and data loss happens.
You can also extend my proposal via D-Bus when a shutdown / reboot / suspend signal is triggered or try something completely different. It was just a suggestion.

**Comment by Alexey-Yakovenko** [comment]
Yes, I understand that this is a suggestion. This was already tried for several years (that's how deadbeef worked originally), and it's a really bad option, that is not coming back.

**Comment by Alexey-Yakovenko** [comment]
Note to self: this is caused by global lock, and the only way to solve it properly is to prevent global locks. Some mechanism is needed to lock whole playlists, and individual tracks separately.
